contracts/Gateway/NativeTokenGateway.sol

```diff
@@ -0,0 +1,232 @@
+     */
+    function wrapAndRepay() external payable nonReentrant {
+        uint256 repayAmount = msg.value;
+        ensureNonzeroValue(repayAmount);
+
+        wNativeToken.deposit{ value: repayAmount }();
+        IERC20(address(wNativeToken)).forceApprove(address(vWNativeToken), repayAmount);
+
+        uint256 borrowBalanceBefore = vWNativeToken.borrowBalanceCurrent(msg.sender);
+        vWNativeToken.repayBorrowBehalf(msg.sender, repayAmount);
+        uint256 borrowBalanceAfter = vWNativeToken.borrowBalanceCurrent(msg.sender);
+
+        IERC20(address(wNativeToken)).forceApprove(address(vWNativeToken), 0);
+
+        if (borrowBalanceAfter == 0 && (repayAmount > borrowBalanceBefore)) {
+            uint256 dust;
+            unchecked {
+                dust = repayAmount - borrowBalanceBefore;
+            }
+
+            wNativeToken.withdraw(dust);
+            _safeTransferNativeTokens(msg.sender, dust);
+        }
+        emit TokensWrappedAndRepaid(msg.sender, address(vWNativeToken), borrowBalanceBefore - borrowBalanceAfter);
+    }
+
+    /**
+     * @notice Sweeps native assets (Native) from the contract and sends them to the owner
+     * @custom:event SweepNative is emitted when assets are swept from the contract
+     * @custom:access Controlled by Governance
+     */
+    function sweepNative() external onlyOwner {
+        uint256 balance = address(this).balance;
+
+        if (balance > 0) {
+            address owner_ = owner();
+            _safeTransferNativeTokens(owner_, balance);
+            emit SweepNative(owner_, balance);
+        }
+    }
+
+    /**
+     * @notice Sweeps the input token address tokens from the contract and sends them to the owner
+     * @param token Address of the token
+     * @custom:event SweepToken emits on success
+     * @custom:access Controlled by Governance
+     */
+    function sweepToken(IERC20 token) external onlyOwner {
+        uint256 balance = token.balanceOf(address(this));
+
+        if (balance > 0) {
+            address owner_ = owner();
+            token.safeTransfer(owner_, balance);
+            emit SweepToken(address(token), owner_, balance);
+        }
+    }
+
+    /**
+     * @dev Redeems tokens, unwrap them to Native Token, and send to the user
+     * This function is internally called by `redeemUnderlyingAndUnwrap` and `redeemAndUnwrap`
+     * @param redeemTokens The amount of tokens to be redeemed. This can refer to either the underlying tokens directly or their equivalent vTokens
+     * @param isUnderlying A boolean flag indicating whether the redemption is for underlying tokens directly (`true`) or for their equivalent vTokens (`false`).
+     * @custom:error ZeroValueNotAllowed is thrown if redeemTokens is zero
+     * @custom:event TokensRedeemedAndUnwrapped is emitted when assets are redeemed from a market and unwrapped
+     */
+    function _redeemAndUnwrap(uint256 redeemTokens, bool isUnderlying) internal {
```

Comment:
Maybe call it `amount` to avoid confusion? `redeemTokens` usually refer to vTokens in the vToken/Comptroller code

```suggestion
    function _redeemAndUnwrap(uint256 amount, bool isUnderlying) internal {
```

Reply:
This `redeemTokens` variable also refers to the vToken amount. Other than that we have deployed the contract on all chains. So, we can leave it as it is :D.